homework/src/index.js

```diff
@@ -34,13 +34,74 @@
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const repositoryDiv = createAndAppend('div', root, { class: 'repositoryDiv' });
+        const header = createAndAppend('header', repositoryDiv, { class: 'topping' });
+        createAndAppend('h1', header, { html: 'HYF Repositories ', class: 'hyfRepo' });
+        const selectBox = createAndAppend('select', header, { class: 'select-box' });
+        data.sort(function (a, b) { return a.name.localeCompare(b.name); });
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selectBox, { value: i, html: data[i].name });
+        }
+        const infoBox = createAndAppend('div', root, { class: 'infoBox' });
+        const ContributorBox = createAndAppend('div', root, { class: 'ContributorBox' });
+        selectBox.addEventListener('change', () => {
+          const value = data[selectBox.selectedIndex].url;
+          fetchJSON(value, (err, data) => {
+            repositoryInfo(infoBox, data, value);
+          });
+          const contributors = data[selectBox.selectedIndex].contributors_url;
+          fetchJSON(contributors, (err, data) => {
+            ContributorsInfo(ContributorBox, data, contributors);
+          });
+        });
+        fetchJSON(data[0].url, (err, data) => {
+          repositoryInfo(infoBox, data);
+        });
+        fetchJSON(data[0].contributors_url, (err, data) => {
+          ContributorsInfo(ContributorBox, data);
+        });
       }
     });
   }
 
+  //adding rows function
+  function addRow(label, value, tbody) {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { html: label });
+    createAndAppend('td', row, { html: value });
+  }
+
+  //information box of each repository
+  function repositoryInfo(infoBox, data) {
+    infoBox.innerHTML = '';
+    const table = createAndAppend('table', infoBox, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    addRow('Repository', data.name, tbody);
+
+    if (data.description) {
+      addRow('Description', data.description, tbody);
+    }
+
+    addRow('Forks', data.forks, tbody);
+    addRow('Updated', data.updated_at, tbody);
+  }
+
+  function ContributorsInfo(ContributorBox, data) {
+    ContributorBox.innerHTML = '';
+    const ContributorTable = createAndAppend('div', ContributorBox, { class: 'ContributorTable' });
+    createAndAppend('p', ContributorTable, { html: 'Contributors', class: 'contTitle' });
+    const ul = createAndAppend('ul', ContributorTable, { class: 'contributorsList' });
+    data.forEach(contributor => {
+      const contributorLink = createAndAppend('a', ul, { href: (`${contributor.html_url}`), target: '_blank' });
```

Comment:
Hi Jim, thank you so much for the notation, i have fixed almost everything now but only the repos that doesnt have contributors, can you help me with a hint what can i do with it...

Reply:
If you would add a `console.log(data);` statement just before line 96 and then, for instance, select the `hyfer-infra` repo (which has no contributors) you would see that the data argument is `null`. In that case you can't do `data.forEach(...)` because `data` is not an array but `null`. That's why you get this error message in the console:

```
index.js:97 Uncaught TypeError: Cannot read property 'forEach' of null
    at ContributorsInfo (index.js:97)
    at fetchJSON (index.js:56)
    at XMLHttpRequest.xhr.onload (index.js:10)
```

So you could test for `null` and only do the `forEach(..)` if `data !== null`:

```js
if (data !== null) {
  data.forEach(..., {
    //...
  });
}
```

P.S. remember to avoid the name `data` if you know what actual data you get, in this case `contributors`.

Try to also fix the remaining comments if you can.